### PR TITLE
Task/recyclerTask

### PR DESCRIPTION
**Pull request by Feinei**
pullReq
pullReq2

### CODE REVIEW

File: app/src/main/java/com/example/homework/AboutActivity.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_about.*
+
+class AboutActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_about)
+
+        val name = intent?.extras?.getString(KEY_NAME) ?: "Unknown"
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/AboutActivity.kt]: "Unknown" вынести в константу типa DEFAULT_STRING or something better

File: app/src/main/java/com/example/homework/Friend.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.example.homework
+
+import android.graphics.drawable.Drawable
+
+data class Friend(
+    var name: String,
+    var friendType: String,
+    var avatar: Drawable
+)
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/Friend.kt]: add single line on file end

File: app/src/main/java/com/example/homework/FriendAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+) : RecyclerView.Adapter<FriendItemHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FriendItemHolder =
+        FriendItemHolder.create(parent, clickLambda)
+
+    override fun getItemCount(): Int = dataSource.size
+
+    override fun onBindViewHolder(holder: FriendItemHolder, position: Int) =
+        holder.bind(dataSource[position])
+}
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/FriendAdapter.kt]: add single line on file end

File: app/src/main/java/com/example/homework/FriendItemHolder.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    companion object {
+
+        fun create(parent: ViewGroup, clickLambda: (Friend) -> Unit) =
+            FriendItemHolder(
+                LayoutInflater.from(parent.context).inflate(R.layout.item_friend, parent, false),
+                clickLambda
+            )
+    }
+}
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/FriendItemHolder.kt]: add single line on file end

File: app/src/main/java/com/example/homework/MainActivity.kt [hunk 5]
```diff
@@ -1,12 +1,41 @@
...
 
 class MainActivity : AppCompatActivity() {
 
+    private var adapter: FriendAdapter? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        adapter = FriendAdapter(getDataSource()) { friend ->
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/MainActivity.kt]: friend в принципе излишен
`
FriendAdapter(getDataSource())
{
navigateToAbout(it)
}
`

File: app/src/main/java/com/example/homework/MainActivity.kt [hunk 6]
```diff
@@ -1,12 +1,41 @@
...
+            navigateToAbout(friend)
+        }
+
+        rv_friends.adapter = adapter
+    }
+
+    private fun navigateToAbout(friend: Friend) {
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/MainActivity.kt]: =

File: app/src/main/java/com/example/homework/MainActivity.kt [hunk 7]
```diff
@@ -1,12 +1,41 @@
...
+        startActivity(AboutActivity.createIntent(this, friend))
     }
+
+    private fun getDataSource(): List<Friend> = listOf(
```
Comment (NailShaykhrazievItis) [inline app/src/main/java/com/example/homework/MainActivity.kt:18]: Лучше вынести в отдельный класс FriendRepository: 
`object FriendRepository {
getDataSource(): List<Friend> = listOf(...)
}`

File: app/src/main/res/layout/activity_about.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:tools="http://schemas.android.com/tools"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".AboutActivity">
+
+    <TextView
+            android:id="@+id/tv_name_title"
+            android:text="Name:"
```
Comment (NailShaykhrazievItis) [inline app/src/main/res/layout/activity_about.xml:12]: ctrl+alt+L
sorting by alphabet

**Review by NailShaykhrazievItis — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_about.xml (hunk 8 above)
Comment (Feinei) [reply app/src/main/res/layout/activity_about.xml:12]: и так всё верно же? no lines changed

Comment (NailShaykhrazievItis) [reply app/src/main/res/layout/activity_about.xml:12]: Нет.
Сначала 
id
style
layout params by alphabet
view parms by  alphabet
app params by  alphabet

**Review by NailShaykhrazievItis — APPROVED** [review]

**Pull request closed without merging**